opta/core/cloud_client.py

```diff
@@ -0,0 +1,24 @@
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
+    from opta.layer import Layer, StructuredConfig
+
+
+class CloudClient:
+    def __init__(self, layer: "Layer"):
+        self.layer = layer
+
+    def get_remote_config(self) -> Optional["StructuredConfig"]:
+        pass
+
+    def upload_opta_config(self) -> None:
+        pass
+
+    def delete_opta_config(self) -> None:
+        pass
+
+    def delete_remote_state(self) -> None:
+        pass
+
+    def get_terraform_lock_id(self) -> str:
+        pass
```

Comment:
```suggestion
    @abstractmethod
    def get_remote_config(self) -> Optional["StructuredConfig"]:
        pass

    @abstractmethod
    def upload_opta_config(self) -> None:
        pass

    @abstractmethod
    def delete_opta_config(self) -> None:
        pass

    @abstractmethod
    def delete_remote_state(self) -> None:
        pass

    @abstractmethod
    def get_terraform_lock_id(self) -> str:
        raise NotImplementedError()
```

- Flag all the methods as abstract to ensure that subclasses override those methods.
- Raise NotImplementedError in `get_terraform_lock` to appease the type checker